### PR TITLE
new prosody module to report census of all rooms

### PR DESCRIPTION
**Pull request by sawall**
add handler for `/room-census` that returns JSON in this form:

```
{
    "room_census": [
        {
            "room_name": "<muc name>",
            "participants": <# participants>,
            "created_time": <unix timestamp>,
        },
       ...
    ]
}
```

### CODE REVIEW

File: resources/prosody-plugins/mod_muc_census.lua [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+--
+function handle_get_room_census(event)
+    local host_session = prosody.hosts["conference." .. tostring(module.host)]
+    if not host_session or not host_session.modules.muc then
+        return { status_code = 400; }
+    end
+
+    room_data = {}
+    for room in host_session.modules.muc.each_room() do
+        if not tostring(room.jid):find("__jicofo-health-check", 1, true) then -- filter out test rooms
```
Comment (sawall) [inline resources/prosody-plugins/mod_muc_census.lua:42]: @damencho is this the best way to filter out the rooms that don't have users? I also notice that they have `nil` as their created time, which could be another way to approach it

Comment (damencho) [reply resources/prosody-plugins/mod_muc_census.lua:42]: Do it like this https://github.com/jitsi/jitsi-meet/blob/2008c9035944e2cc363bc47f72e07529c47289f8/resources/prosody-plugins/mod_muc_allowners.lua#L59

Comment (sawall) [reply resources/prosody-plugins/mod_muc_census.lua]: changed, thanks

File: resources/prosody-plugins/mod_muc_census.lua [hunk 2]
```diff
@@ -0,0 +1,73 @@
+-- report list of rooms with # of members and created date, filtered on test rooms
```
Comment (damencho) [inline resources/prosody-plugins/mod_muc_census.lua:1]: Can you add a description of where this module needs to be loaded, is it under the main virtual host or the muc component?

**Review by damencho — CHANGES_REQUESTED** [review]
This module does not have any protection, we should add some so we avoid anonymous users query and get all rooms and number of participants in it.

**Comment by sawall** [comment]
@damencho : I've improved the module docs in a way that will hopefully address both of the concerns raised.

**Review by damencho — DISMISSED** [review]

**Review by damencho — APPROVED** [review]

**Pull request merged**